src/hdmf/common/table.py

```diff
@@ -218,30 +222,31 @@ def __init__(self, **kwargs):  # noqa: C901
         else:
             id = ElementIdentifiers('id')
 
-        if columns is not None:
-            if len(columns) > 0:
-                # If columns have been passed in, check them over
-                # and process accordingly
-                if isinstance(columns[0], dict):
-                    columns = self.__build_columns(columns)
-                elif not all(isinstance(c, (VectorData, VectorIndex)) for c in columns):
-                    raise ValueError("'columns' must be a list of VectorData, DynamicTableRegion or VectorIndex")
-                colset = {c.name: c for c in columns}
-                for c in columns:
-                    if isinstance(c, VectorIndex):
-                        colset.pop(c.target.name)
-                lens = [len(c) for c in colset.values()]
-                if not all(i == lens[0] for i in lens):
-                    raise ValueError("columns must be the same length")
-                if lens[0] != len(id):
-                    if len(id) > 0:
-                        raise ValueError("must provide same number of ids as length of columns")
-                    else:
-                        id.data.extend(range(lens[0]))
-        else:
```

Comment:
I removed this else because `columns` is checked against None later on and handled there. With this line, `columns` can never be None